linera-core/src/worker.rs

```diff
@@ -334,7 +334,7 @@ where
     #[cfg(test)]
     pub(crate) async fn with_key_pair(mut self, key_pair: Option<Arc<KeyPair>>) -> Self {
         self.chain_worker_config.key_pair = key_pair;
-        self.chain_workers.lock().await.clear();
+        self.chain_workers.lock().unwrap().clear();
```

Comment:
Nit: maybe use `expect` instead?

Reply:
It's standard practice to `unwrap()` `PoisonError`s.  An `expect` here would be something like
```
.expect("the mutex shouldn't be poisoned")
```
which doesn't add much information over the name of the error type :)